Sources/InjiVcRenderer/PreProcessor.swift

```diff
@@ -0,0 +1,172 @@
+import Foundation
+import pixelpass
+
+public class PreProcessor {
+    
+    public init() {}
+    
+    private let QRCODE_IMAGE_TYPE = "data:image/png;base64,"
+
+    public func preProcessVcJson(vcJsonString: String, svgTemplate: String) -> [String: Any] {
+        var vcJsonObject = try! JSONSerialization.jsonObject(with: Data(vcJsonString.utf8), options: []) as! [String: Any]
+        var credentialSubject = vcJsonObject["credentialSubject"] as! [String: Any]
+        credentialSubject = replaceFieldsWithLanguage(credentialSubject)
+
+        // Check for {{qrCodeImage}} for QR Code Replacement
+        if svgTemplate.contains(PreProcessor.QR_CODE_PLACEHOLDER) {
+            let qrCodeImage = replaceQRCode(vcJsonString)
+            credentialSubject[getFieldNameFromPlaceholder(PreProcessor.QR_CODE_PLACEHOLDER)] = qrCodeImage
+        }
+
+        // Check for benefits placeholders
+        let benefitsPlaceholderRegexPattern = PreProcessor.BENEFITS_PLACEHOLDER_REGEX_PATTERN
+        if svgTemplate.range(of: benefitsPlaceholderRegexPattern, options: .regularExpression) != nil {
+            let benefitsPlaceholders = getPlaceholdersList(pattern: benefitsPlaceholderRegexPattern, svgTemplate: svgTemplate)
+            let language = extractLanguageFromPlaceholder(benefitsPlaceholders.first ?? "")
+            let commaSeparatedBenefits = generateCommaSeparatedString(jsonObject: credentialSubject, fieldsToBeCombined: [PreProcessor.BENEFITS_FIELD_NAME], language: language)
+
+            credentialSubject = constructObjectBasedOnCharacterLengthChunks(dataToSplit: commaSeparatedBenefits, placeholderList: benefitsPlaceholders, maxCharacterLength: 55, credentialSubject: credentialSubject, language: language)
```

Comment:
Instead of modifying the `credentialsubject` at multiple places, can we do it once, and keep properly named variables to increase readability and self understandability of code?

Reply:
We need to update the same credentialSubject with the new fields. So only we are updating credentialSubject based on condition